File: backend/music/src/main/java/com/springboot/music/dto/AudioTrackDTO.java
package com.springboot.music.dto;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AudioTrackDTO {

    private Integer id;
    private String title;
    private String audioType;
    private Integer duration;
    private String coverImage;
    private String watermarkedFileUrl;
    private Integer playCount;
    private Double startingPrice;
    private Double averageRating;
    private Long reviewCount;
    private LocalDateTime uploadDate;
    private String status;
    private String authorName;
    private String moderationDecision;
    private String rejectionReason;
    private List<String> revisionPoints;
    private LocalDateTime moderatedAt;
    private String moderatedBy;

    private ArtistDTO artist;
    private TrackTagsDTO tags;

    private List<AudioTrackLicenseDTO> licenses;

    private String description;
    private String lyrics;

}

File: frontend/ReactMusicCommerce/src/responsemodel/AudioTrackDTO.ts
import type { ArtistModel } from "../models/ArtistModel";
import type { AudioTrackLicenseModel } from "../models/AudioTrackLicenseModel";
import type { TrackTagsModel } from "../models/TrackTagsModel";

export interface AudioTrackDTO {
  id: number;
  title: string;
  audioType: string;
  duration: number;
  coverImage: string;
  watermarkedFileUrl: string;
  playCount: number;
  startingPrice: number;
  averageRating: number | null;
  reviewCount: number | null;
  uploadDate: string | null;
  authorName: string;
  artist: ArtistModel;
  tags: TrackTagsModel;
  licenses: AudioTrackLicenseModel[];
  description: string;
}
